packages/esm-patient-chart-app/src/visit/visit-prompt/retrospective-dialog.component.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useCallback, useMemo } from 'react';
+import {
+  Button,
+  ModalBody,
+  ModalHeader,
+  ModalFooter,
+  DatePicker,
+  DatePickerInput,
+  Stack,
+  Row,
+  Column,
+  Form,
+  TimePicker,
+  TimePickerSelect,
+  SelectItem,
+} from '@carbon/react';
+import styles from './retrospective-dialog.scss';
+import { useTranslation } from 'react-i18next';
+import {
+  NewVisitPayload,
+  formatDate,
+  saveVisit,
+  setCurrentVisit,
+  showNotification,
+  toDateObjectStrict,
+  toOmrsIsoString,
+} from '@openmrs/esm-framework';
+import { Controller, FormProvider, useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import dayjs from 'dayjs';
+import { amPm, convertTime12to24 } from '@openmrs/esm-patient-common-lib';
+
+interface RetrospectiveDialogProp {
+  patientUuid: string;
+  closeModal: () => void;
+  onCancel: () => void;
+  visitDate: string;
+  payload: NewVisitPayload;
+}
+
+const RetrospectiveDialog: React.FC<RetrospectiveDialogProp> = ({
+  closeModal,
+  visitDate,
+  payload,
+  patientUuid,
+  onCancel,
+}) => {
+  const { t } = useTranslation();
+
+  const retrospectiveFormSchema = useMemo(() => {
+    return z.object({
+      visitStopDate: z
+        .date({ required_error: t('requiredVisitStopDate', 'Visit stop date is required') })
+        .max(dayjs().subtract(1, 'day').toDate(), {
+          message: t('visitStopDateBeforeCurrentDate', 'Visit stop date should be before current date'),
```

Comment:
Again, I don't think this logic is generalisable across cases. Why can't I enter a retrospective visit that ends today?